pkg/controller/nodeipam/ipam/controller.go

```diff
@@ -71,6 +71,10 @@ func NewController(
 		return nil, fmt.Errorf("invalid IPAM controller mode %q", config.Mode)
 	}
 
+	if containsCIDR(serviceCIDR, clusterCIDR) {
+		return nil, fmt.Errorf("serviceCIDR contains clusterCIDR, node's podCIDR can't be assigned")
+	}
+
```

Comment:
seems like what would be more generic is to check that after occupyServiceCIDR, we have free space left in the clusterCIDR